### PR TITLE
Feat/39 index carousel

### PR DESCRIPTION
**Pull request by mavstronaut**
Fixed base.

Issue [#39](https://github.com/roclin-grp/roclin-grp.github.io/issues/39)

Made some minor tweaks to styling. Still has cleaning up to be done, but in working on this I think I know how I want things to look more clearly.

Obviously, the copy and everything will be focused and honed over time, but I wanted to get some content that might fit the bill to see how the pages would flow approximately.

Played around in paint to work up a logo concept that should do for now. resized it a few times but didn't change the favicons as it is a minor detail for another hour.

The images issue is still an open question, and proper configuration so that the carousel images point to the images folder and not /blog/*.png

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -2,5 +2,151 @@
 # You don't need to edit this file, it's empty on purpose.
```
Comment (loljoho) [inline index.md:2]: @mavstronaut I'm not sure if you saw this comment.

File: index.md [hunk 2]
```diff
@@ -2,5 +2,151 @@
...
 # Edit theme's home layout instead if you wanna make some changes
 # See: https://jekyllrb.com/docs/themes/#overriding-theme-defaults
-layout: home
+title       : Home
+layout      : cover
```
Comment (loljoho) [inline index.md]: Error on serve:
```
Build Warning: Layout 'cover' requested in index.md does not exist.
```

Also, this is how the home page loaded for me:

image

Make sure you're familiar with [YAML Front Matter](https://jekyllrb.com/docs/frontmatter/).

File: _includes/layout/navigation.html [hunk 3]
```diff
@@ -17,12 +17,10 @@
...
         <span class="sr-only">Toggle navigation</span>
         <span class="icon-bar"></span>
-        <span class="icon-bar"></span>
-        <span class="icon-bar"></span>
       </button>
 
-      <a class="navbar-brand" href="index.html">
-        {{ site.title }}
+      <a class="navbar-brand" href="/">
+        <span><img src="/images/rlg-concept-61x65.png"> {{ site.title }}</span>
```
Comment (loljoho) [inline _includes/layout/navigation.html]: Error when navigating to homepage:
```
[2017-12-19 17:19:29] ERROR `/images/rlg-concept-61x65.png' not found.
```

Make sure to commit that image, even if it's not the final version.

File: _includes/layout/footer.html [hunk 4]
```diff
@@ -1,8 +1,59 @@
-<footer class="site-footer">
+<footer class="site__footer">
+
+<script>
```
Comment (loljoho) [inline _includes/layout/footer.html]: This JS code should be located in located in the external `js/main.js` file, as per [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns).

The `_includes/body/scripts.html` partial file is responsible for loading all external scripts.

File: _layouts/contact.html [hunk 5]
```diff
@@ -0,0 +1 @@
+ <div class="typeform-widget" data-url="https://mavrickbarona.typeform.com/to/Uj7res" data-transparency="50" data-hide-headers=true data-hide-footer=true style="width: 100%; height: 100%;" > </div> <script> (function() { var qs,js,q,s,d=document, gi=d.getElementById, ce=d.createElement, gt=d.getElementsByTagName, id="typef_orm", b="https://embed.typeform.com/"; if(!gi.call(d,id)) { js=ce.call(d,"script"); js.id=id; js.src=b+"embed.js"; q=gt.call(d,"script")[0]; q.parentNode.insertBefore(js,q) } })() </script> <div style="font-family: Sans-Serif;font-size: 12px;color: #999;opacity: 0.5; padding-top: 5px;" > powered by <a href="https://www.typeform.com//?utm_campaign=Uj7res&amp;utm_source=typeform.com-11461982-Basic&amp;utm_medium=typeform&amp;utm_content=typeform-embedded-poweredbytypeform&amp;utm_term=EN" style="color: #999" target="_blank">Typeform</a> </div>
```
Comment (loljoho) [inline _layouts/contact.html]: Make sure we discuss any use of third party services, plugins, or modules before implementing them.  I'm not familiar with TypeForm and somehow missed this when I skimmed the commits earlier.

It also renders terribly:

image

For now, a simple [Bootstrap form](https://getbootstrap.com/docs/3.3/css/#forms) will do.  Making a pretty contact form is not that high up on our list of priorities at the moment.

Lastly, this should not be an entire layout file.  Refer to the Jekyll documentation again, especially on [Front Matter](https://jekyllrb.com/docs/frontmatter/) and [Creating pages](https://jekyllrb.com/docs/pages/).

File: index.md [hunk 6]
```diff
@@ -2,5 +2,151 @@
...
+
+
+
+
+<!-- Would like image to maybe be opaque underneath the text. black background for this segment. 
+––––––––––––––––––––––––––––––––––––––––––––––––––  -->
+
+<div class="jumbotron hero-unit">
+  <div class="container" id="about1">
+    <p><img src="/images/rlg-concept2-102x110.png" alt="roclin-logo"></p>
```
Comment (loljoho) [inline index.md]: Error when navigating to page(s):
```
[2017-12-19 17:09:15] ERROR `/images/rlg-concept2-102x110.png' not found.
```

Also, make sure you commit the image, even if it's not the final version.

File: _pages/blog.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                            </div>
+                            <div class="item">
+                                <img src="carousel2.png" alt="roclineage blog">
+                                 <div class="carousel-caption">
+                                     <h1></h1>
+                                     <a class="btn btn-primary btn-lg" href="/blog" role="button">Share your perspective. &raquo;</a> 
+                                 </div>
+                            </div>
+                            <div class="item">
+                              <img src="carousel3.png" alt="work with us">
```
Comment (loljoho) [inline _pages/blog.md]: This should be `` if it's located in the `images/` directory.

Also, commit these images!

File: _pages/blog.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                        <div class="carousel-inner" role="listbox">
+                            <div class="item active">
+                                <img src="carousel1.png" alt="who we are">
+                                <div class="carousel-caption">
+                                    <h1></h1>
+                                    <a class="btn btn-primary btn-lg" href="/about" role="button">Learn about our project. &raquo;</a> 
+                                </div>
+                            </div>
+                            <div class="item">
+                                <img src="carousel2.png" alt="roclineage blog">
```
Comment (loljoho) [inline _pages/blog.md]: This should be `` if it's located in the `images/` directory.

Also, commit these images!

File: _pages/blog.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+                        <ol class="carousel-indicators">
+                            <li data-target="#carousel-feat" data-slide-to="0" class="active"></li>
+                            <li data-target="#carousel-feat" data-slide-to="1"></li>
+                            <li data-target="#carousel-feat" data-slide-to="2"></li>
+                        </ol>
+
+                        <!-- Wrapper for slides -->
+                        <div class="carousel-inner" role="listbox">
+                            <div class="item active">
+                                <img src="carousel1.png" alt="who we are">
```
Comment (loljoho) [inline _pages/blog.md]: This should be `` if it's located in the `images/` directory.

Also, commit these images!

**Review by loljoho — CHANGES_REQUESTED** [review]
Make sure you take the necessary steps to address these issues.  Thanks :)

**Comment by loljoho** [comment]
Oops, I messed up with the Request Review feature of GitHub which is relatively new, and with which I'm not quite familiar yet.

Make sure you look over all the comments I made, and not just the 3 or so that are part of the "requested changes", and push new commits accordingly.

Cheers :)

File: index.md (hunk 1 above)
Comment (mavstronaut) [reply index.md:2]: Saw it before and thought I was putting this somewhere else when I was working. I can move it, but does it really matter in this case? 
Seems like an easy fix but I just looked and couldn't solve it in under 5 minutes so I don't know if it's worth spending time on. This company site only has to work.

File: _includes/layout/footer.html (hunk 4 above)
Comment (mavstronaut) [reply _includes/layout/footer.html]: fixed.

File: index.md (hunk 1 above)
Comment (loljoho) [reply index.md:2]: See [this comment](https://github.com/roclin-grp/roclin-grp.github.io/pull/41#discussion_r157889247).

Sorry about that, all the code comments got split up.

File: _includes/layout/navigation.html (hunk 3 above)
Comment (mavstronaut) [reply _includes/layout/navigation.html]: woop. thought I had, added to commit.

File: _layouts/contact.html (hunk 5 above)
Comment (mavstronaut) [reply _layouts/contact.html]: I am still playing around with it, but it's easy and works. 

I do want the site to be clean and will correct this, but I'm not looking to spend a thousand hours tweaking things on the company site which is just a business card. It only needs to work.

File: _pages/blog.md (hunk 9 above)
Comment (mavstronaut) [reply _pages/blog.md]: Tried this method as you describe and it did not work. Tried again and it still does not work.

File: _pages/blog.md (hunk 8 above)
Comment (mavstronaut) [reply _pages/blog.md]: ** same as other comment. this method did not function.

File: _layouts/contact.html (hunk 5 above)
Comment (loljoho) [reply _layouts/contact.html]: Right, but I'm not sure if it'll work, and adding in a third-party service will complicate things.

And part of the reason I wanted you to work on this is to help you form good practices and coding conventions, as well as learning basic HTML/CSS/JS.

The issues I have with the implementation (as it appears currently) are as follows, off the top of my head:

 - It renders *terribly* in browser and looks unprofessional.  For instance: the overflow (scrolling) is screwed up; the UI is terribly confusing; and the weird background, loading spinner, and navigation disappearing is inconsistent with the rest of the site.
 - I'm not familiar at all with this particular service.  Remember, we still need an AJAX call or *something* here to submit the form data.  
 - Keeping it simple.  Using a third party service for something that can be done with a few lines of HTML, CSS, and JS is overkill.
 - That "Create your own typeform" link in the bottom right is really annoying and ugly.
 - Last time we discussed marketing/email campaign services (e.g. MailChimp) which we would inevitably have to use.  They usually already have form builders, or APIs that are accessible via a simple AJAX call.

File: _pages/blog.md (hunk 7 above)
Comment (mavstronaut) [reply _pages/blog.md]: did not work.

File: index.md (hunk 2 above)
Comment (mavstronaut) [reply index.md]: i have been familiarizing myself as I'm attempting to setup categories for the posts.. however not sure why the landing works for me and not you

Comment (loljoho) [reply index.md]: I was referring to the `layout` parameter of the Front Matter.

File: index.md (hunk 6 above)
Comment (mavstronaut) [reply index.md]: committed.

File: _pages/blog.md (hunk 8 above)
Comment (loljoho) [reply _pages/blog.md]: Okay then.  Commit the images anyways, so I can take a look when you push.

File: _layouts/contact.html (hunk 5 above)
Comment (mavstronaut) [reply _layouts/contact.html]: The form data goes to their service and will look good once it updates. Nothing is required at this point I already tested that I get the submissions. #goodenoughfornow

I'm really not concerned until the point where we are going to start sending out emails, I want to have a working site that can be updated, as the contact form is low priority I can use the mailchimp form submission at that point in time.

Fixing up the copy will be the fun part after these elements are displaying properly.

Comment (loljoho) [reply _layouts/contact.html]: For example: https://kb.mailchimp.com/lists/signup-forms/add-a-signup-form-to-your-website

Adding an additional service *just* for forms seems like overkill to me.

Comment (mavstronaut) [reply _layouts/contact.html]: it'd make sense talking on it in conversation. features need to be operational now. they can be reworked again later as things are done over. 
i very much want to get both of us focused on the tagmap visualization and work on the company page as i'm writing posts and our marketing / branding / product come together.

**Comment by loljoho** [comment]
I only see one new commit `1b09a26`.  Are you still working on this?

**Comment by loljoho** [comment]
Eschewing strict and proper workflow to get this finished.

Closes: #39

**Pull request merged**